build/lib/android/index.js

```diff
@@ -27,25 +28,32 @@ class Android {
 		this.apiLevel = options.apiLevel;
 		this.sdkVersion = options.sdkVersion;
 		this.gitHash = options.gitHash;
+		this.sdk = new AndroidSDK(this.androidSDK, this.apiLevel);
+		this.antProperties = {
+			'build.version': this.sdkVersion,
+			'build.githash': this.gitHash,
+			'android.sdk': this.sdk.getAndroidSDK(),
+			'android.platform': this.sdk.getPlatformDir(),
+			'google.apis': this.sdk.getGoogleApisDir(),
+			'kroll.v8.build.x86': 1,
+			'android.ndk': this.androidNDK
+		};
 	}
 
 	async clean() {
-		return ant.build(ANDROID_BUILD_XML, [ 'clean' ], {});
+		return ant.build(ANDROID_BUILD_XML, [ 'clean' ], this.antProperties);
 	}
 
 	async build() {
-		const AndroidSDK = require('./androidsdk');
-		const sdk = new AndroidSDK(this.androidSDK, this.apiLevel);
-		const properties = {
-			'build.version': this.sdkVersion,
-			'build.githash': this.gitHash,
-			'android.sdk': sdk.getAndroidSDK(),
-			'android.platform': sdk.getPlatformDir(),
-			'google.apis': sdk.getGoogleApisDir(),
-			'kroll.v8.build.x86': 1,
-			'android.ndk': this.androidNDK
-		};
-		return ant.build(ANDROID_BUILD_XML, [ 'full.build' ], properties);
+		// Clean build and download V8
+		await this.clean().catch(error => console.error(error));
```

Comment:
really don't need the catch chained here, it'll bubble up the Error if there is one. adding a catch will actually just log it and move on.